tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,387 @@
+        self.count = self.count + 1
+        raise self.error
+
+    def PollGenerateReport(self):
+        self.count = self.count + 1
+        raise self.error
+
+    def SubmitGenerateReport(self, report_request):
+        self.count = self.count + 1
+        raise self.error
+
+    def PollGenerateReport(self, request_id):
+        self.count = self.count + 1
+        raise self.error
+
+    @property
+    def factory(self):
+        self.count = self.count + 1
+        raise self.error
+
+    @property
+    def soap_client(self):
+        self.count = self.count + 1
+        raise self.error
+
+    @property
+    def call_count(self):
+        return self.count
+
+@mock.patch("tap_bing_ads.filter_selected_fields_many", return_value = '')
+@mock.patch("singer.write_records", return_value = '')
+@mock.patch("singer.metrics", return_value = '')
+@mock.patch("singer.write_bookmark", return_value = '')
+@mock.patch("singer.write_state", return_value = '')
+@mock.patch("tap_bing_ads.sobject_to_dict", return_value = '')
+@mock.patch("singer.get_bookmark", return_value = {'b1': 'b1'})
+@mock.patch("singer.write_schema", return_value = '')
+@mock.patch("tap_bing_ads.get_core_schema", return_value = '')
+@mock.patch("tap_bing_ads.get_selected_fields", return_value = '')
+class TestBackoffError(unittest.TestCase):
+    '''
+    Test that backoff logic works properly. Mocked some common method to test the backoff including
+    filter_selected_fields_many, write_records , metrics, write_bookmark, write_state, sobject_to_dict,
+    get_bookmark, write_schema, get_core_schema, get_selected_fields, time.sleep.
+    '''
+    @mock.patch("tap_bing_ads.create_sdk_client", return_value = '')
+    @mock.patch("tap_bing_ads.CustomServiceClient")
+    def test_url_error_get_account(self, mock_get_account, mock_create_sdk_client,
+                                                    mock_get_selected_fields, mock_get_core_schema,
+                                                    mock_write_schema, mock_get_bookmark,
+                                                    mock_sobject_to_dict, mock_write_state,
+                                                    mock_write_bookmark, mock_metrics, mock_write_records,
+                                                    mock_filter_selected_fields_many):
+        '''
+        Test that tap retry on the url timeout error for 60 seconds.
+        '''
+        mock_get_account.side_effect = URLError('_ssl.c:1059: The handshake operation timed out')
+        before_time = datetime.datetime.now()
+        try:
+            tap_bing_ads.sync_accounts_stream(['i1'], {})
+        except URLError:
+            pass
+        after_time = datetime.datetime.now()
+        time_difference = (after_time - before_time).total_seconds()
+        # verify the code backed off for 60 seconds
+        # time_difference should be greater or equall 60 as some time elapsed while calculating `after_time`
```

Comment:
```suggestion
        # time_difference should be greater or equal 60 as some time elapsed while calculating `after_time`
```
This typo is present at multiple places so replace it with `equal`.

Reply:
done